packages/ui/src/components/ManageChannels/ManageChannelsScreenView.tsx

```diff
@@ -330,9 +330,40 @@ export function ManageChannelsScreenView({
         type: c.type,
       })),
     }));
-
-    // Only update if the total number of sections have changed
+    const currentTotalChannels = sections.reduce(
+      (acc, section) => acc + section.channels.length,
+      0
+    );
+    const newTotalChannels = newNavSections.reduce(
+      (acc, section) => acc + section.channels.length,
+      0
+    );
+
+    // Only update local state if the total number of sections have changed
+    // OR if a newly created channel has been added to the default section
+    // OR if a channel has been deleted
     if (newNavSections.length === sections.length) {
```

Comment:
Based on the comment, it seems like we'd wanna run those inner IF's even if this conditional is false (no new section added)?

Reply:
Not sure what you mean, this condition would only be false if a new section was added. We're trying to catch cases where a new section wasn't added but we need to know if a channel was added or deleted.

Reply:
That makes sense, I understand the intention now.